miden/src/examples/mod.rs

```diff
@@ -49,10 +58,13 @@ pub enum ExampleType {
 }
 
 impl ExampleOptions {
-    pub fn get_proof_options(&self) -> ProvingOptions {
+    pub fn get_proof_options(&self) -> Result<ProvingOptions, ExecutionOptionsError> {
+        let exec_options = ExecutionOptions::new(Some(self.max_cycles), self.expected_cycles)?;
         match self.security.as_str() {
-            "96bits" => ProvingOptions::with_96_bit_security(self.recursive),
-            "128bits" => ProvingOptions::with_128_bit_security(self.recursive),
+            "96bits" => Ok(ProvingOptions::with_96_bit_security(self.recursive)
+                .with_execution_options(exec_options)),
+            "128bits" => Ok(ProvingOptions::with_128_bit_security(self.recursive)
+                .with_execution_options(exec_options)),
             other => panic!("{} is not a valid security level", other),
         }
```

Comment:
minor nit: this could also be done as:
```Rust
match self.security.as_str() {
    "96bits" => Ok(ProvingOptions::with_96_bit_security(self.recursive),
    "128bits" => Ok(ProvingOptions::with_128_bit_security(self.recursive)
    other => panic!("{} is not a valid security level", other),
}.with_execution_options(exec_options))
```

Reply:
This implementation is not working for me, maybe I have an old version of rust. For me this approach is working:
```rust
Ok(match self.security.as_str() {
    "96bits" => ProvingOptions::with_96_bit_security(self.recursive),
    "128bits" => ProvingOptions::with_128_bit_security(self.recursive),
    other => panic!("{} is not a valid security setting", other),
}
.with_execution_options(exec_options))
```
but it looks strange and could be confusing.

Reply:
No worries - it's fine to keep as you had it.